src/DropdownToggle.js

```diff
@@ -55,14 +55,17 @@ class DropdownToggle extends React.Component {
       ...props
     } = this.props;
 
+    if (Component === DropdownToggle.defaultProps.as) {
+      props.bsPrefix = childBsPrefix;
```

Comment:
how about we only assign this if it's not `undefined`? less fragile than this check, and would fix the problem at hand

Reply:
In that case I was thinking we'd run into this issue again if somone passes both `as` and `childBsPrefix`, but since it's a private prop after all it doesn't need to be supported ?

Reply:
The problem isn't `as` existing, it's that childBsPrefix isn't defined but still passed through. So it's enough to check if childBsPrefix is not undefined before assigning it to props. If it is defined then someone is trying to ass it through

Reply:
if you do specify it and your `as` doesn't take that component, then... yeah that's just a bug, i'm fine w/not doing anything clever there